iothub/service/src/DirectMethod/DirectMethodRequest.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Parameters to execute a direct method on the device
+    /// </summary>
+    public class DirectMethodRequest
+    {
+        /// <summary>
+        /// Method to run
+        /// </summary>
+        [JsonProperty("methodName", Required = Required.Always)]
+        public string MethodName { get; set; }
+
+        /// <summary>
+        /// The timeout (in seconds) before the direct method request will fail if the device doesn't respond to the request.
+        /// This timeout may happen if the target device is slow in handling the direct method.
+        /// </summary>
+        [JsonProperty("responseTimeoutInSeconds", NullValueHandling = NullValueHandling.Ignore)]
+        public int? ResponseTimeout { get; set; }
```

Comment:
Let's make this a TimeSpan.

https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/iothub/service/src/Jobs/JobRequest.cs#L64-L75

Reply:
Also ConnectionTimeout